src/meta/src/barrier/progress.rs

```diff
@@ -247,35 +279,38 @@ impl std::fmt::Debug for TrackingJob {
                 "TrackingJob::New({:?})",
                 command.context.table_to_create()
             ),
-            TrackingJob::Recovered(recovered) => {
+            TrackingJob::RecoveredV1(recovered) => {
                 write!(
                     f,
-                    "TrackingJob::Recovered({:?})",
+                    "TrackingJob::RecoveredV1({:?})",
                     recovered.fragments.table_id()
                 )
             }
+            TrackingJob::RecoveredV2(recovered) => {
+                write!(f, "TrackingJob::RecoveredV2({:?})", recovered.id)
+            }
         }
     }
 }
 
-pub struct RecoveredTrackingJob {
+pub struct RecoveredTrackingJobV1 {
```

Comment:
Do we still need `TrackingCommand` in the enum above?  I think the work is the same for all job no matter it's from a new command or a recovered one. So I think we may just store the information needed by either v1 or v2, and don't need such extra enum.

Besides I think we don't need to store the `metadata_manager` per job, because they should always be the same `metadata_manager`. We can instead store it in the global progress tracker.

Reply:
Perhaps. It will require much more refactoring though. I think it can be done in a separate PR.